enhancements/network/infiniband.md

```diff
@@ -0,0 +1,123 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+RDMA (Remote Direct Memory Access) fabrics such as Infiniband and RDMA over
+Converged Ethernet (RoCE) are key to unlocking scalable performance for
+demanding Artificial Intelligence (AI) and Machine Learning (ML) applications.
+RoCE v2 is supported in SR-IOV Operator as Tech Preview since 4.3, This
+enhancement is to introduce native Infiniband support.
+
+## Motivation
+
+Enable high-performance message transmission via RDMA for distributed workloads
+such as AI/ML.
+
+### Goals
+
+- Provide native Infiniband support on pod additional network interfaces
+
+### Non-Goals
+
+- Support RoCE v2
+- Support GPUDirect RDMA
+- Support Ethernet over Infiniband such as IPoIB (IP over Infiniband)
+- Support Infiniband on pod primary network interface
+
+## Proposal
+
+Extend SR-IOV Operator to support both Ethernet and Infiniband devices.
+
+SR-IOV Hardware NIC is managed by SR-IOV Operator and will be provisioned
+with a new device type `infiniband` or `ib`.
+
+A new Container Network Interface (CNI) plugin `ib-sriov-cni` will be added.
+
+Infiniband support will only be available on bare metal deployment.
+
+#### Hardware
+
+The following Mellanox NICs will be supported:
+- Mellanox Technologies MT27700 Family [ConnectX-4]
+- Mellanox Technologies MT28908 Family [ConnectX-6]
+
+#### Infiniband SR-IOV CNI
+
+Infiniband SR-IOV CNI `ib-sriov-cni` is a 3rd party CNI plugin that moves
+SR-IOV Infiniband device to pod namespace and configures device attributes
+such as Infiniband GUID. It will be delivered in a separate image and copied
+to node CNI directory by a DaemonSet.
+
+#### SR-IOV Config Daemon
+
+SR-IOV Config Daemon, a sub component managed by SR-IOV Operator, will be
+enhanced to change device link type to Infiniband.
+
+#### SR-IOV Operator
+
+SR-IOV Operator API definition will support `infiniband` or `ib` device type
+in both `SriovNodeNetworkState` and `SriovNodeNetworkPolicy`.
+
+## Design Details
+
+### Test Plan
+
+- e2e tests will be added in SR-IOV Operator for Infiniband device type
+- e2e tests will be conducted in baremetal environment as part of SR-IOV tests
+- Infiniband function must work on supported NICs
+
+### Graduation Criteria
+
+##### -> Tech Preview
+
+- SR-IOV Infiniband device can be provisioned by SR-IOV Operator
+- SR-IOV Infiniband device can be advertised as extended node resource
+- Pod can be created with SR-IOV Infiniband device as additional interface
+- SR-IOV document should be updated to include Infiniband feature
+- Should pass e2e and QE tests
+
+### Upgrade / Downgrade Strategy
+
+This enhancement is an addon feature in SR-IOV Operator, it follows the same
+upgrade/downgrade strategy as was developed in marketplace operator.
+
+## Implementation History
+
+- SR-IOV Operator is in GA support since 4.3
+- RoCE v2 support in SR-IOV Operator is Tech Preview since 4.3
```

Comment:
what preventing RoCE to be GA?

Reply:
I think we need users' feedback on the feature so that we can evaluate whether it is mature enough to be GA.

Reply:
Right, feedback and more testing.